app/src/marketplace/containers/EditProductPage2/DeployCommunityModal.jsx

```diff
@@ -42,39 +44,65 @@ function setSkipGuide(value) {
     storage.setItem(SKIP_GUIDE_KEY, JSON.stringify(value))
 }
 
+// allow 60s for the API to start in CP server
+const API_READY_ESTIMATE = 60
+
 export const DeployDialog = ({ product, api, updateAddress }: DeployDialogProps) => {
     const dontShowAgain = skipGuide()
     const [step, setStep] = useState(dontShowAgain ? steps.CONFIRM : steps.GUIDE)
+    const [deployError, setDeployError] = useState(null)
+    const [estimate, setEstimate] = useState(0)
     const [address, setAddress] = useState(null)
     const dispatch = useDispatch()
+    const isMounted = useIsMounted()
 
     const onClose = useCallback(() => {
         api.close(!!address && isEthereumAddress(address))
     }, [api, address])
+    const onCloseRef = useRef()
+    onCloseRef.current = onClose
 
     const productId = product.id
     // $FlowFixMe
     const { adminFee = 0 } = product || {}
     const onDeploy = useCallback(async () => {
         const { id: joinPartStreamId } = await createJoinPartStream(productId)
 
+        if (!isMounted()) { return Promise.resolve() }
+
+        // Set estimate
+        const blockEstimate = await averageBlockTime(getWeb3())
+        setEstimate(blockEstimate + API_READY_ESTIMATE)
```

Comment:
Do we need to add `API_READY_ESTIMATE` here? API is spinning up while the transaction is being mined. Only case for this could be that `blockEstimate` is really short but for the time being it isn't :)

Reply:
Yep, I shortened it to 5s now.